test/e2e/network/dns_configmap.go

```diff
@@ -185,11 +185,51 @@ func (t *dnsNameserverTest) run() {
 		moreForeverTestTimeout)
 }
 
+type dnsPtrFwdTest struct {
+	dnsTestCommon
+}
+
+func (t *dnsPtrFwdTest) run() {
+	t.init()
+
+	t.createUtilPod()
+	defer t.deleteUtilPod()
+
+	t.createDNSServerWithPtrRecord()
+	defer t.deleteDNSServerPod()
+
+	t.setConfigMap(&v1.ConfigMap{Data: map[string]string{
+		"upstreamNameservers": fmt.Sprintf(`["%v"]`, t.dnsServerPod.Status.PodIP),
+	}})
+
+	moreForeverTestTimeout := 2 * 60 * time.Second
+
+	t.checkDNSRecordFrom(
+		"123.2.0.192.in-addr.arpa",
+		func(actual []string) bool { return len(actual) == 1 && actual[0] == "my.test." },
+		"dnsmasq",
+		moreForeverTestTimeout)
+
+	t.c.CoreV1().ConfigMaps(t.ns).Delete(t.name, nil)
+	t.checkDNSRecordFrom(
+		"123.2.0.192.in-addr.arpa",
+		func(actual []string) bool { return len(actual) == 0 },
+		"dnsmasq",
+		moreForeverTestTimeout)
+}
+
 var _ = SIGDescribe("DNS configMap nameserver", func() {
-	t := &dnsNameserverTest{dnsTestCommon: newDnsTestCommon()}
-	BeforeEach(func() { t.c = t.f.ClientSet })
+
+	nsTest := &dnsNameserverTest{dnsTestCommon: newDnsTestCommon()}
+	fwdTest := &dnsPtrFwdTest{dnsTestCommon: newDnsTestCommon()}
```

Comment:
Why not move these into It block as well?

Reply:
Done. The test above was also changed by the way.